Add multiple tutorial pages covering the "Registration" material from the SCT course (#3094)
spinalcordtoolbox/spinalcordtoolbox: documentation/source/user_section/tutorials/customizing_registration.rst

* **Step 0:** Straighten the spinal cord, then match the subject labels to the template labels.
* **Step 1:** Nonrigid deformation, first pass. Deals with large deformations in the spinal cord.
* **Step 2:** Nonrigid deformation, second pass. Applies fine cord shape adjustments.

Typically, step 0 is not altered. However, Steps 1 and 2 can be tweaked, and additional steps (e.g. 3, 4) can be added. Some of the common parameters to tweak include:

   .. figure:: https://raw.githubusercontent.com/spinalcordtoolbox/doc-figures/master/registration_to_template/sct_register_to_template-param-algo.png
      :align: right
      :figwidth: 40%

      Visualization of algorithms to choose from for the ``algo`` parameter of ``-param``.

* ``algo`` This is the algorithm used to compute the nonrigid deformation. Choice of algorithm depends on how coarse/fine you want your transformation to be. This depends on which step you are modifying (Step 1, step 2, step 3, etc.) as well as the nature of the spinal cord you are working with.

   - **translation**: axial translation (x-y)

>    - **rigid**: transaltion + rotation about z axis

Reviewer: ```suggestion
   - **rigid**: x-y translation + rotation about z axis
```

   - **affine**:
   - **b-splinesyn**: based on ants binaries
   - **syn**: (not bspline regularized)

>    - **slicereg**: ANTs+Us, regualarized translation across slices that is regularized across S-I (used with segmentation for a pre-alignment cord centerline + template cord centerline -- basically alignment of the core centerline)

Reviewer: ```suggestion
   - **slicereg**: slice-wise translations within the axial plane. The translations within each slice are regularized across slices in the superior-inferior (S-I) direction. This algorithm can be used to align two images that are already "close" to each other, or it can be used to align two segmentations for a pre-alignment of cord centerline. 
```

>    - **centermassrot**: similar to slicereg, center of mass of segmentation on each slice and then align with center of mass in the template space, also does a rotation in case your subject turned neck or sometimes compression

Reviewer: ```suggestion
   - **centermassrot**: Compute the center of mass of the source and destination segmentation and align them slice-by-slice. This algorithm also estimates the orientation of the cord in the axial plane and matches that of the destination cord. This feature could be useful in case the subject rotated their neck, or if the cord is rotated due to a disc compression. Note that this algorithm only works with segmentations (contrary to slicereg). 
```

>    - **columnwise**: in case of highly compressed cords (suggested by UofT Alan Martin) it basically takes the segmentation and tries to match the compressed segmentation with the cord segmentation of the template. nonlinear deformation with much more degree of freedom than the syn based approaches

Reviewer: ```suggestion
   - **columnwise**: This algorithm is particularly interested in case of highly compressed cords. The first step consists in matching the edges of the source and destination segmentations in the R-L direction (scaling operation), and during the second step, each row of the source segmentation is matched to the destination segmentation (scaling operation in the A-P direction). After iterating across all rows in the R-L direction, a warping field is produced. This non-linear deformation is more controlled than the SyN-based approach. This method only works with segmentations (not images). The idea came from Dr. Allan Martin (University of Toronto, UC Davis). 
```

> * ``type``: Carefully chose type={im, seg} based on the quality of your data, and the similarity with the template. Ideally, you would always choose type=im . However, if you find that there are artifacts of image features (e.g., no CSF/cord contrast) that could compromise the registration, then use type=seg instead. Of course, if you choose type=seg , make sure your segmentation is good (manually adjust it if it is not). By default, the sct_register_to_template relies on the segmentations only because it was found to be more robust to the existing variety of MRIs. This last step that is basically a slicewise nonlinear deformation there are a lot of algorithms that are available in SCT. Those algorithms can either be run on the image or could be run on the segmentation. Highly artifacted EPI the csf is missing on one slice, appearing on other slice, distortions, DIPS artifacts, etc. You know that your segmetnation is correct because you manually corrected it, you might want to rely on the segmentation mroe than the image in order to register your spinal cord to the template (there is a possibility to do that)

Reviewer: not sure what happened here-- copy/paste included another chunk of text
```suggestion
* ``type``: Carefully choose ``type={im, seg}`` based on the quality of your data, and the similarity with the template. Ideally, you would always choose ``type=im``. However, if you find that there are artifacts of image features (e.g., no CSF/cord contrast) that could compromise the registration, then use ``type=seg`` instead. Of course, if you choose ``type=seg``, make sure your segmentation is good (manually adjust it if it is not). By default, ``sct_register_to_template`` relies on the segmentations only because it was found to be more robust to the existing variety of MRIs. 
```

> * ``metric``: Adjust metric based on type. With type=im , use metric=CC (accurate but long) or MI (fast, but requires enough voxels) With type=seg , use metric=MeanSquares.

Reviewer: ```suggestion
* ``metric``: Adjust metric based on type. With ``type=im``, use ``metric=CC`` (cross-correlation: accurate but long) or ``MI`` (mutual information: fast, but requires enough voxels). With ``type=seg`` or with images with the exact same contrast and intensity (e.g., fMRI time series, or two images acquired with the same acquisition parameters), use ``metric=MeanSquares``.
```

> * ``slicewise``: All transformations are constrained in Z direction, though estimation can be done slice-wise or volume-wise: -param slicewise={0, 1}

Reviewer: ```suggestion
* ``slicewise``: Only applies to ``algo={translation, rigid, affine, syn, bsplinesyn}``. If set to ``0``, a unique 3D transformation is estimated. If set to ``1``, transformations are estimated for each axial slice independently. 
```
